### PR TITLE
Replace Span enum with struct.

### PR DESCRIPTION
**Pull request by sebpuetz**
Makes life easier when spans are mutated, e.g. going from
continuous span to skip span is much more straight forward.

### CODE REVIEW

File: lumberjack/src/span.rs [hunk 1]
```diff
@@ -4,76 +4,89 @@ use std::ops::Range;
...
+    /// Create new span with skipped indices.
+    ///
+    /// `skips` has to be non-empty, to construct a span without skips, use `Span::new`.
+    pub(crate) fn new_with_skips(lower: usize, upper: usize, skips: HashSet<usize>) -> Self {
+        assert!(lower < upper, "Span start has to be smaller then end.");
+        assert!(
+            !skips.is_empty(),
+            "Skips have to be non-empty if using this constructor."
+        );
+        assert_ne!(upper - lower - skips.len(), 0, "Can't skip all indices.");
```
Comment (twuebi) [inline lumberjack/src/span.rs]: What if skips.len() > (upper-lower?)

File: lumberjack/src/span.rs [hunk 2]
```diff
@@ -4,76 +4,89 @@ use std::ops::Range;
...
+        (self.start, self.end)
     }
 
     /// Get the number of indices covered.
     pub fn n_indices(&self) -> usize {
-        match self {
-            Span::Discontinuous(span) => span.upper - span.lower - span.skip.len(),
-            Span::Continuous(span) => span.upper - span.lower,
+        if let Some(ref skips) = self.skips {
+            self.end - self.start - skips.len()
```
Comment (twuebi) [inline lumberjack/src/span.rs:81]: Is this subject to changes, or could `n_indices` be set at construction time?

File: lumberjack/src/span.rs [hunk 3]
```diff
@@ -92,134 +105,36 @@ impl Span {
...
     }
 }
 
 impl Ord for Span {
     fn cmp(&self, other: &Span) -> Ordering {
-        if self.lower() != other.lower() {
-            self.lower().cmp(&other.lower())
-        } else if self.upper() != other.upper() {
-            self.upper().cmp(&other.upper())
+        if self.start != other.start {
```
Comment (twuebi) [inline lumberjack/src/span.rs:134]: Maybe add a comment detailing how the spans are ordered.

File: lumberjack/src/span.rs (hunk 2 above)
Comment (sebpuetz) [reply lumberjack/src/span.rs:81]: Spans can/will change

**Comment by sebpuetz** [comment]
Danke!

**Pull request merged**